src/vm/i386/asmhelpers.S

```diff
@@ -836,13 +836,14 @@ NESTED_ENTRY VirtualMethodFixupStub, _TEXT, NoHandler
     // Set frame pointer
     PROLOG_END
 
-    sub     esp, 8
-    push    ebx         // address of the thunk
+    push    eax
+    sub     esp, 4
+    push    eax         // address of the thunk
     push    ecx         // this ptr
-
     CHECK_STACK_ALIGNMENT
     call    C_FUNC(VirtualMethodFixupWorker)
-    add     esp, 8
+    add     esp, 4
+    pop     eax
```

Comment:
The address of the thunk is not needed after the call to VirtualMethodFixupWorker.